File: flink-job-example-no-flink-lib.py
from kafka import KafkaConsumer

client_id = 'example-flink-job-no-lib'
shared_topic = 'flink-topic'

def process_message(value):
    # Example processing logic
    return value.upper()

def run_job():
    consumer = KafkaConsumer(
        shared_topic,
        bootstrap_servers='localhost:9092',
        group_id='flink-group',
        auto_offset_reset='earliest'
    )

    print('Job started. Press Ctrl+C to exit.')

    for message in consumer:
        value = message.value.decode('utf-8')
        print(f'Received message: {value}')

        # Process the message
        processed_value = process_message(value)
        print(f'Processed message: {processed_value}')

def shutdown():
    print('Shutting down job...')
    exit(0)

if __name__ == '__main__':
    try:
        run_job()
    except KeyboardInterrupt:
        shutdown()

File: flink-job-example-no-flink-lib.js
const { Kafka } = require('kafkajs');
const readline = require('readline');

const clientId = 'example-flink-job-no-lib';
const sharedTopic = 'flink-topic';

const kafka = new Kafka({
  brokers: ['localhost:9092'],
  clientId: clientId,
});

const consumer = kafka.consumer({ groupId: 'flink-group' });

// Set up readline interface
readline.emitKeypressEvents(process.stdin);
process.stdin.setRawMode(true);

async function runJob() {
  await consumer.connect();
  await consumer.subscribe({ topic: sharedTopic, fromBeginning: true });

  console.log('Job started. Press Ctrl+C to exit.');

  await consumer.run({
    eachMessage: async ({ topic, partition, message }) => {
      const value = message.value.toString();
      console.log(`Received message: ${value}`);

      // Process the message
      const processedValue = processMessage(value);
      console.log(`Processed message: ${processedValue}`);
    },
  });
}

function processMessage(value) {
  // Example processing logic
  return value.toUpperCase();
}

async function shutdown() {
  console.log('Shutting down job...');
  await consumer.disconnect();
  process.exit(0);
}

runJob().catch(console.error);

process.on('SIGINT', shutdown);
